pvlib/iam.py

```diff
@@ -748,3 +750,255 @@ def marion_integrate(function, surface_tilt, region, num=None):
+
+    Parameters
+    ----------
+    surface_tilt : numeric
+        Surface tilt angle measured from horizontal (e.g. surface facing
+        up = 0, surface facing horizon = 90). [degrees]
+
+    Returns
+    -------
+    iam_sky : numeric
+        The incident angle modifier for sky diffuse
+
+    iam_ground : numeric
+        The incident angle modifier for ground-reflected diffuse
+
+    References
+    ----------
+    .. [1] Schlick, C. An inexpensive BRDF model for physically-based
+       rendering. Computer graphics forum 13 (1994).
+
+    .. [2] Xie, Y., M. Sengupta, A. Habte, A. Andreas, "The 'Fresnel Equations'
+       for Diffuse radiation on Inclined photovoltaic Surfaces (FEDIS)",
+       Renewable and Sustainable Energy Reviews, vol. 161, 112362. June 2022.
+       :doi:`10.1016/j.rser.2022.112362`
+
+    See Also
+    --------
+    pvlib.iam.schlick
+    pvlib.iam.fedis_diffuse
+    """
+    # these calculations are as in [2]_, but with the refractive index
+    # weighting coefficient w set to 1.0 (so it is omitted)
+
+    # relative transmittance of sky diffuse radiation by PV cover:
+    cosB = cosd(surface_tilt)
+    sinB = sind(surface_tilt)
+    cuk = (2 / (np.pi * (1 + cosB))) * (
+        (30/7)*np.pi - (160/21)*np.radians(surface_tilt) - (10/3)*np.pi*cosB
+        + (160/21)*cosB*sinB - (5/3)*np.pi*cosB*sinB**2 + (20/7)*cosB*sinB**3
+        - (5/16)*np.pi*cosB*sinB**4 + (16/105)*cosB*sinB**5
+    )  # Eq 4 in [2]
+
+    # relative transmittance of ground-reflected radiation by PV cover:
+    with np.errstate(divide='ignore', invalid='ignore'):  # Eq 6 in [2]
+        cug = 40 / (21 * (1 - cosB)) - (1 + cosB) / (1 - cosB) * cuk
+
+    cug = np.where(surface_tilt < 1e-6, 0, cug)
+
+    # respect input types:
+    if np.isscalar(surface_tilt):
+        cuk = cuk.item()
+        cug = cug.item()
+    elif isinstance(surface_tilt, pd.Series):
+        cuk = pd.Series(cuk, surface_tilt.index)
+        cug = pd.Series(cug, surface_tilt.index)
+
+    return cuk, cug
+
+
+def fedis(aoi, n=1.5, n_ref=None):
+    """
+    Determine the incidence angle modifier (IAM) for direct irradiance
+    using the FEDIS transmittance model.
+
+    Note that FEDIS [1]_ calculates direct IAM using the Fresnel equations
+    without extinction, thus in the default case of ``n_ref=None``, this is
+    equivalent to calling :py:func:`physical` with ``K=0``.
+
+    Parameters
+    ----------
+    aoi : numeric
+        Angle of incidence. [degrees]
+
+    n : float, default 1.5
+        Refractive index of the PV front surface material.  The default value
+        of 1.5 was used for an IMT reference cell in [1]_. [unitless]
+
+    n_ref : float, optional
+        Reference refractive index. If None (default), set equal to ``n``.
+        In [1]_ ``n_ref`` was set to 1.4585 for
+        a fused silica dome over a CMP22, but in conventional
+        PV applications it is appropriate to set ``n_ref=n`` (the default
+        behavior).
```

Comment:
The explanation of n_ref is not really clear enough.  I suggest two concrete examples, to illustrate the need for sometimes having them equal and sometimes not.  (Unfortunately I don't understand the author's thinking on this.)

Reply:
I agree this description is not very useful.  What do you think of this revision?

```
    n_ref : float, optional
        Refractive index for which IAM=1.0 at normal incidence.
        If None (default), set equal to ``n``. See Notes for
        possible use cases. [unitless]
```

And the following in `Notes`:
```
        In most PV applications it is appropriate to set ``n_ref=n``
        (the default behavior), but a custom ``n_ref`` may be useful when
        using irradiance data from a pyranometer whose calibration does not
        account for reflection off its glass dome.  It may also be used as
        the original index of refraction if the module surface's refractive
        index is changing over time (for example a degraded AR coating).
```

Reply:
> may be useful when
>         using irradiance data from a pyranometer whose calibration does not
>         account for reflection off its glass dome.

In other words: never.

Reply:
I think the author needs to provide this information.

Reply:
Pending author input, I think the revised description is clear.

Reply:
Author input has been incorporated with a minor addition besides the revision proposed above.  Any other thoughts here @adriesse?

Reply:
I think the two use cases are still inadequately described.  This is why I suggested getting an example, which would be concrete; and another reviewer, who might evaluate this from a less biased perspective.